App.tsx

```diff
@@ -4,19 +4,164 @@ import { ImageModal } from './ImageModal';
 import { DetailsPage } from './DetailsPage';
 import { createStackNavigator } from '@react-navigation/stack';
 import { NavigationContainer } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { WeatherApp } from './WeatherApp/Weather';
+import { WeatherStackParam } from './WeatherApp/WeatherStackParam';
+import { ScannerScreen } from './BarCodeScanner/BarCodeScannerScreen';
+import { ScannerStackParam } from './BarCodeScanner/ScannerStackParam';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import FavoritesPage from './BarCodeScanner/FavoritesPage';
+import { Ionicons } from '@expo/vector-icons';
+import { ProductDetails } from './BarCodeScanner/ProductDetails';
+import Battery from './Battery/Battery';
+import { CharCreator } from './Final/CharCreator';
+import { CharViewer } from './Final/CharViewer';
+import { CharModal } from './Final/CharModal';
 
 export type StackParamList = {
   Gallery: undefined,
-  Details: { id: number; url: string};
+  Details: { id: number; url: string},
   Modal: {id: number; url: string},
 }
 
 const Stack = createStackNavigator();
+const Weather = createStackNavigator<WeatherStackParam>();
+const Scanner = createBottomTabNavigator<ScannerStackParam>();
+const Drawer = createDrawerNavigator();
+const Character = createBottomTabNavigator();
+const CharStack = createStackNavigator();
 
-export default function App() {
+const WeatherNav = () => {
   return (
-    <NavigationContainer>
-      <Stack.Navigator initialRouteName='Gallery'>
+    <Weather.Navigator>
+      <Weather.Screen name="main" component={WeatherApp}/>
+    </Weather.Navigator>
+  )
+}
+
+const Tab = createBottomTabNavigator();
+
+function CharViewerStack()
+{
+  return (
+    <CharStack.Navigator>
+      <CharStack.Screen 
+      name='CharViewer' 
+      component={CharViewer} 
+      options={{
+        headerShown: false,
+      }} />
+      <CharStack.Screen options={{
+        presentation: "modal",
+        title: "Character Details",
+        headerStyle: {
+          backgroundColor: "black",
+        },
+        headerTintColor: "white",
+        }} name='CharModal' component={CharModal} />
+    </CharStack.Navigator>
+  )
+};
+
+function ProductDetailsStack() 
+{
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="main" 
+      component={ScannerScreen}
+      options={{
+        headerShown: false,
+        headerTitle: 'Bar-Code Scanner',
+        headerStyle: { backgroundColor: "#838081", borderBottomColor: 'black',},
+      }}
+      />
+      <Stack.Screen name="ProductDetails" component={ProductDetails} />
+    </Stack.Navigator>
+  )
+};
+
+const CharacterNav = () => {
+  return (
+    <Character.Navigator initialRouteName="CharCreator" 
+    tabBarOptions={{
```

Comment:
This property is deprecated. you should be using `screenOptions` instead. This warning is displayed each time you open the app.